core/src/main/java/com/scalar/db/api/CrudOperable.java

```diff
@@ -131,4 +155,31 @@ public interface CrudOperable<E extends TransactionException> {
    * @throws E if the transaction CRUD operation fails
    */
   void mutate(List<? extends Mutation> mutations) throws E;
+
+  /** A scanner abstraction for iterating results. */
+  interface Scanner<E extends TransactionException> extends AutoCloseable, Iterable<Result> {
+    /**
+     * Returns the first result in the results.
+     *
+     * @return the first result in the results
+     * @throws E if the operation fails
+     */
+    Optional<Result> one() throws E;
+
+    /**
+     * Returns all the results.
+     *
+     * @return the list of {@code Result}s
+     * @throws E if the operation fails
+     */
```

Comment:
Likewise to the comment above. I think the Javadoc should be more clear that it iterates over the results.

```suggestion
  /**
   * Returns all remaining results.
   *
   * @return a List containing all remaining results
   * @throws E if the operation fails
   */
```

Reply:
Fixed in 8ae4db17b493762e66bef5657863c499daac7540. Thanks!